tests_fp/solidity/test_add_liquidity_fp.py

```diff
@@ -84,85 +94,85 @@ def test_add_liquidity_with_long_immediately(self):
-        self.assertEqual(True, True)
         lp_supply_before = self.hyperdrive.market_state.lp_total_supply
 
         # Celine opens a short.
         market_deltas, wallet_deltas = self.hyperdrive.open_short(
             agent_wallet=self.celine.wallet,
-            bond_amount=50_000_000,
+            bond_amount=FixedPoint("50_000_000.0"),
         )
 
         # Add liquidity with the same amount as the original contribution.
         market_deltas, wallet_deltas = self.hyperdrive.add_liquidity(self.bob.wallet, self.contribution)
 
         # Ensure that the contribution was transferred to Hyperdrive.
-        self.assertEqual(market_deltas.d_base_asset, -wallet_deltas.balance.amount)
+        self.assertEqual(float(market_deltas.d_base_asset), float(-wallet_deltas.balance.amount))
 
         # Ensure that the new LP receives the same amount of LP shares as the initializer.
-        self.assertAlmostEqual(market_deltas.d_lp_total_supply, lp_supply_before, 6)
-        self.assertEqual(self.hyperdrive.market_state.lp_total_supply, lp_supply_before * 2)
+        self.assertAlmostEqual(float(market_deltas.d_lp_total_supply), float(lp_supply_before), places=6)
+        self.assertEqual(float(self.hyperdrive.market_state.lp_total_supply), float(lp_supply_before) * 2)
 
         # Ensure the pool APR is still approximately equal to the target APR.
         pool_apr = self.hyperdrive.pricing_model.calc_apr_from_reserves(
             self.hyperdrive.market_state, self.hyperdrive.position_duration
         )
-        self.assertAlmostEqual(pool_apr, self.target_apr, 1)
+        self.assertAlmostEqual(float(pool_apr), float(self.target_apr), places=1)
 
     def test_add_liquidity_with_long_at_maturity(self):
         """Test adding liquidity with a long at maturity."""
         # Celine opens a long.
         market_deltas, wallet_deltas = self.hyperdrive.open_long(
             agent_wallet=self.celine.wallet,
-            base_amount=50_000_000,
+            base_amount=FixedPoint("50_000_000.0"),
         )
 
-        self.block_time.tick(1)
+        # Advance 1 year
+        self.block_time.tick(FixedPoint("1.0"))
 
         # Add liquidity with the same amount as the original contribution.
         market_deltas, wallet_deltas = self.hyperdrive.add_liquidity(self.bob.wallet, self.contribution)
 
         # Ensure that the contribution was transferred to Hyperdrive.
-        self.assertEqual(market_deltas.d_base_asset, -wallet_deltas.balance.amount)
+        self.assertEqual(float(market_deltas.d_base_asset), float(-wallet_deltas.balance.amount))
 
         # Ensure the pool APR is still approximately equal to the target APR.
         pool_apr = self.hyperdrive.pricing_model.calc_apr_from_reserves(
             self.hyperdrive.market_state, self.hyperdrive.position_duration
         )
-        self.assertAlmostEqual(pool_apr, self.target_apr, 1)
+        self.assertAlmostEqual(float(pool_apr), float(self.target_apr), places=1)
```

Comment:
1 place... it was like this already but let's try like 3 places.

Reply:
maxed out precision wherever I could. Still not great